backend_plugin/src/main/evaluators/BlockCommentEvaluator.java

```diff
@@ -29,72 +34,52 @@ public BlockCommentEvaluator() {
 
-						// If we have two back slashes, then check if we've found two previously
-						if (firstBackSlashDetected) {
-							IRegion currentRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
+						// Check if consecutive lines
+						int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
+						int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
 
-							// DEBUG
-							// System.out.println("Current Region pos: " + currentRegion.getOffset());
-							// System.out.println("Current REgion length: " + currentRegion.getLength());
-							// System.out.println("Prev Region pos: " + prevRegion.getOffset());
-							// System.out.println("Prev Region length: " + prevRegion.getLength());
-
-							// Check if consecutive lines
-							int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
-							int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
-
-							// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
-							if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
-									(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
-								firstBackSlashDetected = false;
-								return true;
-							}
-
-							// If not consecutive, then update prevRegion
+						// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
+						if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
+								(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
 							prevRegion = currentRegion;
-
-							// If we haven't seen a double back slash yet
-						} else {
-							firstBackSlashDetected = true;
-							prevRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
+							return true;
 						}
+
+						// If not consecutive, then update prevRegion
+						prevRegion = currentRegion;
+
+						// If we haven't seen a double back slash yet
+					} else {
+						firstBackSlashDetected = true;
+						prevRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
 					}
-				// Try Catch needed in case offset passed in doc.get methods doesn't actually exist
-				// We shouldn't actually get here, but since it's all asynchronous, something weird may happen
-				} catch (BadLocationException e) {
-					// DEBUG
-					System.out.println("Bad offset location in BlockCommentEvaluator"); // shouldn't be able to get here
-					firstBackSlashDetected = false;
 				}
+			// Try Catch needed in case offset passed in doc.get methods doesn't actually exist
+			// We shouldn't actually get here, but since it's all asynchronous, something weird may happen
+			} catch (BadLocationException e) {
+				// DEBUG
+				System.out.println("Bad offset location in BlockCommentEvaluator"); // shouldn't be able to get here
```

Comment:
Unless we expect something catastrophic to happen if the offset doesn't exist, we don't need the console log here. If we do expect that, then we should probably do more than just log it to the console (e.g. shut down the listener)